### PR TITLE
CRW-4585: fix CVE issues

### PR DESCRIPTION
**Pull request by svor**
This PR provides a few updates to fix CVE issues:

CVE-2023-34034 - by updating **spring-security-config** from 6.1.0 to 6.1.4
CVE-2021-23926 - by updating **org.apache.xmlbeans:xmlbeans** from 2.6.0 to 5.1.1
CVE-2022-1471 - by updating **org.yaml:snakeyaml** from 1.3 to 2.0

Related issue: https://issues.redhat.com/browse/CRW-4585

### CODE REVIEW

**Review by nickboldt — APPROVED** [review]
LGTM, did not test, and so won't merge this on your behalf.

File: server/build.gradle [hunk 1]
```diff
@@ -64,6 +68,7 @@ configurations {
 }
 
 dependencies {
+    implementation "org.apache.xmlbeans:xmlbeans:${versions.xmlbeans}"
     implementation "org.springframework.boot:spring-boot-starter-web"
```
Comment (nickboldt) [inline server/build.gradle:72]: should these springboot versions be pinned to 3.1.4 too?

Comment (svor) [reply server/build.gradle:72]: it's enough to set the version of springboot in plugins block

**Review by SDawley — APPROVED** [review]
Looks good. Checked the output of ./gradlew dependencies and they all are updated.

**Pull request merged**